### ISSUE TITLE
Snack bar fires visually whenever a value changes. 

### ISSUE THREAD

**Issue by jennlove-um** [tl-its-umich-edu/my-learning-analytics#1589]
Snack bar fires visually whenever a value changes.  If I am using the arrow keys in any input[type="number"] it will fire with every increase.

Paradoxically the firings are not voiced by assistive tech (probably bc the snackbar is wrapped in a role="presentation") which is a good thing. Maybe fire onblur instead and voice it? Whatever - retain the role="presentation" if not planning to change how the snackbar fires - otherwise it would be too distracting for screen reader users.

### Steps to Reproduce :
1. Go to Assignment Planning
2. In Set Minimum Goal field, use the arrow key to raise or lower the goal grade.  The snack bar fires with each click.
3. Once a minimum goal is set, go to an assignment and use the arrow key to raise or lower the goal grade. The snack bar fires with each click. 

[MyLA Technical Review 7/24 - row 4](https://docs.google.com/spreadsheets/d/186xZEvxhbh55jLR_mRXT9tMrmFUlcBlUtkN5T7BuT64/edit?gid=886470260#gid=886470260)

**Comment by jonespm** [comment]
It looks like this is on an onChange now. I agree it would probably work better being on an onBlur event instead. Voicing this message could be something worth doing then too but might be separate.

**Comment by jaydonkrooss** [comment]
My concern with using `onBlur` is that the snackbar wouldn't fire until after user activity, which doesn't reflect the true time that the setting was saved (during `onChange`)

I've been looking into the option to instead use `debounce` from the lodash package, which we're already utilizing for the overall grade minimum input field. I discussed this with @gsilver ; the solution would be to debounce the snackbar so it only fires once within 500ms of the most recent change. We could customize the timing of this, but it should reduce the number of times the snackbar opens when the field is edited within a short time period. 

The other idea discussed in the MyLA meeting today would be to remove the use of the number field altogether and instead use a text input with tighter restriction/validation on text. The motivation for this stems from [this article](https://stackoverflow.blog/2022/12/26/why-the-number-input-is-the-worst-input/), but I suspect this may take a lot more component tinkering.

**Comment by jaydonkrooss** [comment]
Test plan: modify two different inputs on the page - the overall goal grade field and the individual assignment goal field.
Overall grade goal field: write once 
1. Modify the grade goal number field (either type/paste in a value or use the built-in increment buttons)
2. There will be a short duration (250ms) after the change before the screen updates the bar visualization and re-writes the various assignment fields below
3. Then there is a duration (750ms) after which the snackbar showing the settings saved notification. For a11y checks, Make sure this alerts screen readers

Overall grade goal field: write multiple times with debounce
1. Modify the grade goal number field 
2. Within 1 second, change the number field again, either by typing or using the built-in increment button (up and down arrow-keys for keyboard users)
3. The shorter and longer duration (250ms + 750ms) listed above will commence immediately after this change, after which you will see **only one** snackbar notification

Individual assignment field: write once 
1. Modify any assignment grade number field 
2. Note that these events happen immediately: "Lock goal" checkbox, if not already checked, will be checked and multiple recalculations of other assignments will cause the fields to be rewritten. 
3. After a duration (750 ms), the snackbar notification will appear

Individual assignment field: write multiple times with debounce 
1. Modify any assignment grade number field 
2. Within 750ms, modify it again either with new values typed via keyboard or with the built-in arrows (up and down keys for the keyboard)
3. After a duration (750 ms) of the second change, **only one** snackbar notification will appear

**Comment by gsilver** [comment]
Hey Jaydon, lovely test plan, it is going up on the hall of fame. 

Ship it! The toast fires after a decent interval - this will benefit everyone, really, but specially neurodivergent and blind users. Nice work

**Comment by jaydonkrooss** [comment]
Glad to hear the interval is working well. By the way, just to correct an observation made on the screen-reader aspect to this,  I've discovered that the reason why the message was not registering to the screen reader is that the child component had role="alertdialog", but we weren't adding the right attributes to label and focus the element. I don't think the "presentation" role had anything to do with this.

After I removed this, the default went to role="alert", which seemed to be read in Voiceover appropriately. I know we're not supposed to rely on MacOS Voiceover, but I think this was just a sanity-check for me, and I hope we can verify this with a proper screen-reader test on this change soon.
